src/PuliExtension.php

```diff
@@ -99,7 +100,7 @@ public function getTokenParsers()
     public function getFunctions()
     {
         if (!$this->urlGenerator) {
-            return array();
+            throw new RuntimeException('The resource_url() Twig function requires a UrlGenerator to be set in PuliExtension.');
```

Comment:
This solution is causing the extension to fail anytime that it is loaded without the URL generator. I think it would be better to add a public `generateUrl()` function here and do the check in there. Then `resource_url()` can be mapped to that function. The result is that the exception will only be thrown when the `resource_url()` function is actually used.